test/unit/conditionals.spec.ts

```diff
@@ -112,3 +112,29 @@ test.each([false, true, null])("Ternary conditional with generic whenTrue branch
         })
         .expectToMatchJsResult();
 });
+
+test.each([false, true])("Ternary conditional with preceding statements in true branch (%p)", trueVal => {
+    // language=TypeScript
+    util.testFunction`
+        let i = 0;
+        const result = ${trueVal} ? i += 1 : i;
+        return { result, i };
+    `
+        .setOptions({
+            strictNullChecks: true,
```

Comment:
Why is this option required here, I don't see any undefineds/nulls in the tests?

Reply:
Without it, the test doesn't start as failing and test the issue:
The type `number` with no strict null checks may be null and therefore falsy, so the if statement is used.